src/internal/observable/interval.ts

```diff
@@ -52,20 +54,17 @@ import { Subscriber } from '../Subscriber';
  * @name interval
  * @owner Observable
  */
-export function interval(period = 0,
-                         scheduler: SchedulerLike = async): Observable<number> {
+export function interval(period = 0, scheduler: SchedulerLike = asyncScheduler): Observable<number> {
   if (!isNumeric(period) || period < 0) {
     period = 0;
   }
 
-  if (!scheduler || typeof scheduler.schedule !== 'function') {
-    scheduler = async;
+  if (!isScheduler(scheduler)) {
+    scheduler = asyncScheduler;
   }
 
-  return new Observable<number>(subscriber => {
-    subscriber.add(
-      scheduler.schedule(dispatch as any, period, { subscriber, counter: 0, period })
-    );
+  return new Observable<number>((subscriber) => {
+    subscriber.add(scheduler.schedule(dispatch as any, period, { subscriber, counter: 0, period }));
     return subscriber;
```

Comment:
issue: Unrelated to the change, but I don't think it's necessary to return the subscriber here.